file-formats/rvbc/type/zif_aff_rvbc_v1.intf.abap

```diff
@@ -0,0 +1,93 @@
+"! AFF type for the object RVBC ( Review Booklets )
+INTERFACE zif_aff_rvbc_v1
+  PUBLIC.
+  "! $values { @link zif_aff_rvbc_v1.data:co_extensibility_mode }
+  "! $default { @link zif_aff_rvbc_v1.data:co_extensibility_mode.not_supported }
+  TYPES ty_extensibility_mode TYPE n LENGTH 2.
+  TYPES ty_app_id             TYPE c LENGTH 70.
+  TYPES ty_source             TYPE n LENGTH 1.
+  TYPES ty_status             TYPE n LENGTH 1.
+  TYPES ty_consistency_status TYPE n LENGTH 1.
+  TYPES ty_ina1_service_id    TYPE c LENGTH 30.
+  TYPES ty_ina1_service_name  TYPE c LENGTH 40.
+  TYPES ty_ina1_service_type  TYPE n LENGTH 2.
+
+  CONSTANTS:
+    BEGIN OF co_extensibility_mode,
+      "! <p class="shorttext">Fully Supported Including Custom Analytical Queries</p>
+      "! Fully supported including custom analytical queries
+      fully_supported              TYPE ty_extensibility_mode VALUE 0,
+      "! <p class="shorttext">Supported Only for Predefined Analytical Queries</p>
+      "! Supported only for predefined analytical queries
+      supported_predefined_queries TYPE ty_extensibility_mode VALUE 1,
+      "! <p class="shorttext">Not Supported</p>
+      "! Not supported
+      not_supported                TYPE ty_extensibility_mode VALUE 2,
+    END OF co_extensibility_mode.
+
+
+  TYPES:
+    "! <p class="shorttext">Review Booklet Definition</p>
+    "! Review booklet definition
+    BEGIN OF ty_booklet_definition,
+      "! <p class="shorttext">Application ID</p>
+      "! Application ID
+      "! $required
+      application        TYPE ty_app_id,
+      "! <p class="shorttext">Source</p>
+      "! Source
+      "! $required
+      source             TYPE ty_source,
+      "! <p class="shorttext">Status</p>
+      "! Status
+      "! $required
+      status             TYPE ty_status,
+      "! <p class="shorttext">Consistency Status</p>
+      "! Consistency status
+      "! $required
+      consistency_status TYPE ty_consistency_status,
+      "! <p class="shorttext">Extensibility Mode</p>
+      "! Extensibility mode
+      "! $required
+      extensibility_mode TYPE ty_extensibility_mode,
+    END OF ty_booklet_definition.
+
+  TYPES:
+    "! <p class="shorttext">InA Service</p>
+    "! InA Service
+    BEGIN OF ty_ina1_service,
+      "! <p class="shorttext">Service ID</p>
+      "! Service ID
+      "! $required
+      ina1_service_id TYPE ty_ina1_service_id,
```

Comment:
From what I understood in our call, this id will be filled during the RVBC creation (generated with an ADT Generator), together with an InA Service. This field will be hidden from the user, and will it ever be updated? For example when the ina1_service field is changed?

Additionally, since the ina1_service is actually the name of a service binding object, and these have unique names, now I am a bit confused about the service_id. Which object/property had non-unique names?

When we talked about this, I hadn't understood that the ina1_service was actually of type service binding, and not the InA1 object type, thus why I am having these questions now.

Reply:
the ina1_service_id is the DB key, therefore it will not be updated when user change e.g. ina1_service. However it is needed to be carried all the way, even it is not visible for the ADT editor UI, for the purpose that during workbench integration we can easily fetch the correct DB instance based on the ina1_service_id key. The workbench integration will be on top of our RAP implementations, including logics like actions, validations, determinations, authority checks and so on. All RAP related statements are using EML language and are working on entity key(DB key). that is why we have to carry it all the time, indepdent of the uniqueness of other fields.

Reply:
we tool your proposal and remove the ID field and type field now.